airflow/jobs/scheduler_job.py

```diff
@@ -695,6 +695,7 @@ def _process_task_instances(self, dag, task_instances_list, session=None):
         # update the state of the previously active dag runs
         dag_runs = DagRun.find(dag_id=dag.dag_id, state=State.RUNNING, session=session)
         active_dag_runs = []
+        dag_run_finished_ti_map = {}
```

Comment:
I don't think we need to store this collection - it doesn't seem we look at this again, so the existing `finished_task` being inside the per-dagrun loop is enough?